BoxContentSDK/BoxContentSDK/Requests/BOXSearchRequest.m

```diff
@@ -28,75 +34,100 @@ - (instancetype)initWithSearchQuery:(NSString *)query inRange:(NSRange)range
     return self;
 }
 
+- (instancetype)initWithTemplateKey:(NSString *)templateKey scope:(NSString *)scope filters:(NSArray *)filters inRange:(NSRange)range
+{
+    BOXAssert(templateKey, @"TemplateKey must be non-nil for BOXSearchRequest");
+    BOXAssert(scope, @"Scope must be non-nil for BOXSearchRequest");
+    BOXAssert(&range, @"Range must be non-nil for BOXSearchRequest");
+    
+    if (self = [self initWithSearchQuery:nil inRange:range]) {
+        self.templateKey = templateKey;
+        self.scope = scope;
+        self.filters = filters ? filters : @[];
+    }
+    
+    return self;
+}
+
 - (BOXAPIOperation *)createOperation
 {
+    BOOL isMetadataSearch = self.templateKey && self.scope && self.filters;
     NSURL *URL = [self URLWithResource:BOXAPIResourceSearch
                                     ID:nil
                            subresource:nil
                                  subID:nil];
     
     NSMutableDictionary *queryParameters = [NSMutableDictionary dictionary];
-    queryParameters[BOXAPIParameterKeyQuery] = self.query;
-
-    if (self.requestAllItemFields) {
-        queryParameters[BOXAPIParameterKeyFields] = [self fullItemFieldsParameterString];
-    }
-    
-    if (self.fileExtensions) {
-        queryParameters[BOXAPIParameterKeyFileExtensions] = [self.fileExtensions componentsJoinedByString:@","];
+    NSString *queryKey = nil;
+    if (isMetadataSearch) {
+        self.query = [self generateMetadataQuery];
+        queryKey = BOXAPIParameterKeyMDFilter;
+    } else {
+        queryKey = BOXAPIParameterKeyQuery;
     }
-
-    // If both to and from values are not set, the API allows trailing commas (e.g. "value," or ",value")
-    if (self.createdAtFromDate || self.createdAtToDate) {
-        NSString *fromDate = [self.createdAtFromDate box_ISO8601String];
-        NSString *toDate = [self.createdAtToDate box_ISO8601String];
-        queryParameters[BOXAPIParameterKeyCreatedAtRange] =
+    queryParameters[queryKey] = self.query;
+
+    if (!isMetadataSearch) {
+        if (self.requestAllItemFields) {
```

Comment:
aren't these applicable to the metadata search too? I'd think request all item fields would be...